src/pages/GuestInfo.tsx

```diff
@@ -0,0 +1,159 @@
+import { Box, FormField, TextInput, DateInput, Form, Button } from 'grommet';
+import { useNavigate } from 'react-router-dom';
+import { PageWrapper } from './PageWrapper';
+import Logger from '../utils/logger';
+import { useCallback, useState } from 'react';
+import axios from 'axios';
+import { MessageBox } from 'src/components/MessageBox';
+import { DateTime } from 'luxon';
+import { useAppDispatch, useAppState } from '../store';
+import { PersonalInfo } from 'src/store/types';
+import { backend } from '../config';
+
+// const offerId = 1;
+const logger = Logger('GuestInfo');
+const defaultValue = {
+  firstname: '',
+  lastname: '',
+  birthday: new Date(),
+  email: '',
+  phone: ''
+};
+
+export const GuestInfo = () => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const { checkout } = useAppState();
+
+  const [value, setValue] = useState<PersonalInfo>(defaultValue);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<undefined | string>();
+
+  const handleSubmit = useCallback(async () => {
+    try {
+      setError(undefined);
+      setLoading(true);
+      if (
+        value.firstname === '' ||
```

Comment:
Many form libaries allows you to pass to an input whether it is required or not. If it is empty, you cannot submit the field.

Is that possible with this system that you are using for forms? Then you don't have to do check manually for every value like you are doing now.

Reply:
@bobetbat this is absolutely wrong way. The Grommet form has `validators` that can be attached to form fields. It is very flexible.

Reply:
✔️